doc(release) Release notes for OSS and Collect 24.10.0 (#3721)
centreon/centreon-documentation: i18n/fr/docusaurus-plugin-content-docs/version-24.10/releases/centreon-os.mdx

Pour faire des demandes d'évolutions ou signaler des bugs sur les extensions commerciales, vous pouvez vous rendre sur
notre [Github](https://github.com/centreon/centreon/issues/new/choose).

> Retrouvez plus de détails sur la version 24.10 dans notre [post de blog](https://www.centreon.com/fr/centreon-annonce-la-sortie-de-sa-derniere-version-logicielle-centreon-24-10/).

<ExpandCollapseAll />

## Centreon Web

### 24.10.0

Release date: `October 30, 2024`
 
<details open>
  <summary>Enhancements</summary>

> - [ACL] Improved host groups listing based on Resources Access configuration.

Reviewer: ```suggestion
- [ACL] Non-admin users can now see immediately freshly-created host groups in the list of host groups.
```

> - [ACL] Improved service groups listing based on Resources Access configuration.

Reviewer: ```suggestion
- [ACL] Non-admin users can now see freshly-created service groups immediately in the list of service groups.
```

- [API] Added an endpoint that automatically deploys the services brought by the host template.
- [API] Added an endpoint to list Pollers.
- [API] Added an endpoint to create Broker inputs.
- [API] Added an endpoint to create Additional Connector Configurations.
- [API] Added an endpoint to delete Additional Connector Configurations.
- [API] Added an endpoint to update an Additional Connector Configuration.
- [API] Added an endpoint to list/get an Additional Connector Configuration.

> - [API] Added API endpoints to return the complete hierarchy for a given BA

Reviewer 2: ```suggestion
- [API] Added API endpoints to return the complete hierarchy for a given BA.
```

- [API] Added configuration change logs for hostgroup endpoints.
- [API] Added missing configuration change logs for host configuration.
- [Authentication] Expired tokens are now deleted faster.
- [Authentication] You can now enable/disable API Tokens.

> - [CEIP] Added telemetry for Additional Connector Configuration.

Reviewer: ```suggestion
```

- [CLAPI] CLAPI now manages multiline values.
- [Configuration] Added the ability to duplicate a locked command.
- [Configuration] You can now configure several more Broker loggers from the interface.

> - [Dashboards] Added a pending refresh indicator to widgets

Reviewer 2: ```suggestion
- [Dashboards] Added a pending refresh indicator to widgets.
```

- [Dataset] When users select a resource, only the data that is linked to this resource is displayed in the lists for the other types of data. Irrelevant entries are filtered out.
- [Metrics Graph widget] Added curve type format setting.
- [Metrics Graph widget] Added multiple options to customize graph.
- [Metrics Graph widget] Distinguished curves for same metric name, by automatically adjusting colors.
- [Metrics Graph widget] Enabled visualization using bar chart and stacked bar chart.
- [Resource Status] In the filter window, you can now filter by resource type.
- [Resource Status] Users can now  filter by host name in view by host mode.
- [Resource Table widget] Added ticketing (by third-party provider) feature.
- [Resource Table widget] Added action buttons, similar to Resource Status (Acknowledge, Downtime, Forced check).
- [Resource Table widget] Added filtering options by severity and by status type.
- [Resource Table widget] Moved display type selection into the widget.
- [Secrets] Manage Vault credentials storage/update/duplicate/delete for Host/Service/Host template/Service template and snmp community macros.

> - [Status Grid widget] Added Centreon resources pictograms

Reviewer 2: ```suggestion
- [Status Grid widget] Added Centreon resources pictograms.
```

> - [Status Grid widget] Added colored banners for resources states

Reviewer 2: ```suggestion
- [Status Grid widget] Added colored banners for resources states.
```

- [UX] Removed autocomplete for credentials fields in administration pages.

> - [Web Page widget] Implemented new widget to display a web page, using its URL.

Reviewer 2: ```suggestion
- [Web Page widget] Implemented new widget to display a web page using its URL.
```

</details>

<details>
  <summary>Bug fixes</summary>

- [API] Fixed access control for hosts configuration endpoints.
- [API] Fixed access control for meta-service endpoint.
- [API] Fixed access control for monitoring servers configuration endpoints.
- [API] Fixed access control for notification configuration endpoints.

> - [API] Fixed access control for some v2 API endpoints.
> - [ACL] Fixed access rights for dashboard configuration endpoints.
> - [API] Fixed access control for some v2 API endpoints.

Reviewer: ```suggestion
- [API] Fixed access control for some v2 API endpoints.
- [ACL] Fixed access rights for dashboard configuration endpoints.
```

- [Configuration] Added an option that allows users to disable checks for a service whose host is down.
- [Engine] Engine can now handle extra configuration files to complement/overload the centengine.cfg file so that custom configurations remain unchanged when the standard configuration is exported.
- [Engine] Added a new centengine parameter (`send_recovery_notifications_anyways`) to force sending a recovery notification even when the resource is outside of its notification period. This aims at keeping third-party tools up-to-date at any time when they are fed notifications.
- [Engine] The whitelist now works with all the commands executed by Engine, including notifications and event handlers.
- [gRPC API] The GetHost function has been improved to return more information.
- [Packaging] Collect packages are now also available for Debian 12 (bookworm) ARM64 (for pollers on Raspberry Pi).

</details>

## Centreon Monitoring Agent

<details open>
  <summary>Enhancements</summary>

> - [CMA] Added a silent mode (CLI) to the agent's Windows installer.
> - [CMA] Centreon Monitoring Agent is now available with a Windows installer, which installs the service and deploys Centreon plugins on the host.

Reviewer: ```suggestion
- [CMA] Centreon Monitoring Agent is now available with a Windows installer, which installs the service and deploys Centreon plugins on the host.
- [CMA] Added a silent mode (CLI) to the agent's Windows installer.
```

- Added missing configuration change logs for host configuration.
- CLAPI now manages multiline values.

**Authentication**

- Expired tokens are now deleted faster.
- You can now enable/disable API Tokens.

**Configuration**

- You can now duplicate a locked command.
- You can now configure several more Broker loggers from the interface.

**Dashboards and widgets**

- Added last update time to widgets and enabled refresh directly from inside widgets.
- You can now create a public link to a playlist.
- Status chart and status grid widgets: when users filter by service / service group / service category and choose to display the data by host, the tiles corresponding to the hosts are displayed correctly instead of displaying "no host found".
- Allowed smaller dimensions for widgets through more responsive behavior.
- Optimized data payload required by widgets to retrieve resource data.
- When users select a resource, only the data that is linked to this resource is displayed in the lists for the other types of data. Irrelevant entries are filtered out.
- **New widgets**:
   - The Web Page widget displays a web page using its URL.

>    - "Clock / Timer" widget.

Reviewer: ```suggestion
   -  The "Clock / Timer" widget displays the time according to the selected time zone, or a timer.
```

  <summary>Enhancements</summary>

**ACL**

- Non-admin users can now see freshly-created host groups and service groups in the list of host/service groups immediately.

**Additional Connector Configurations**

- The **Additional connector configurations** menu allows you to define the credentials to access a monitored resource through the Centreon interface rather than having to connect to your pollers manually.

**API**

- Added an endpoint that automatically deploys the services brought by the host template.
- Added an endpoint to list Pollers.
- Added an endpoint to create Broker inputs.
- Added endpoints to create, delete, update and list/get Additional Connector Configurations.
- Added API endpoints to return the complete hierarchy for a given BA.
- Added configuration change logs for hostgroup endpoints.
- Added missing configuration change logs for host configuration.
- CLAPI now manages multiline values.

**Authentication**

- Expired tokens are now deleted faster.
- You can now enable/disable API Tokens.

**Configuration**

- You can now duplicate a locked command.
- You can now configure several more Broker loggers from the interface.

**Dashboards and widgets**

> - Added a "pending refresh" indicator to widgets.
> - Added last update time to widgets and enabled refresh directly from inside widgets.

Reviewer: ```suggestion
- Added last update time to widgets and enabled refresh directly from inside widgets.
```

- You can now create a public link to a playlist.

> - Added the display of parent hosts statuses when only services are selected in status chart and status grid widgets.

Reviewer: ```suggestion
- Status chart and status grid widgets: when users filter by service / service group / service category and choose to display the data by host, the tiles corresponding to the hosts are displayed correctly instead of displaying "no host found".
```

Pour faire des demandes d'évolutions ou signaler des bugs sur les extensions commerciales, vous pouvez vous rendre sur
notre [Github](https://github.com/centreon/centreon/issues/new/choose).

> Retrouvez plus de détails sur la version 24.10 dans notre [post de blog](https://www.centreon.com/fr/centreon-annonce-la-sortie-de-sa-derniere-version-logicielle-centreon-24-10/).

<ExpandCollapseAll />

## Centreon Web

### 24.10.0

Release date: `October 30, 2024`
 
<details open>
  <summary>Enhancements</summary>

**ACL**

> - Non-admin users can now see immediately freshly-created host groups in the list of host groups.

Reviewer: ```suggestion
- Non-admin users can now see freshly-created host groups in the list of host groups immediately.
```

> - Non-admin users can now see freshly-created service groups immediately in the list of service groups.

Reviewer: ```suggestion
- Non-admin users can now see freshly-created service groups in the list of service groups immediately.
```

- Added configuration change logs for hostgroup endpoints.
- Added missing configuration change logs for host configuration.
- CLAPI now manages multiline values.

**Authentication**

- Expired tokens are now deleted faster.
- You can now enable/disable API Tokens.

**Configuration**

- You can now duplicate a locked command.
- You can now configure several more Broker loggers from the interface.

**Dashboards and widgets**

- Added a "pending refresh" indicator to widgets.
- Added last update time to widgets and enabled refresh directly from inside widgets.
- You can now create a public link to a playlist.
- Added the display of parent hosts statuses when only services are selected in status chart and status grid widgets.
- Allowed smaller dimensions for widgets through more responsive behavior.
- Optimized data payload required by widgets to retrieve resource data.
- When users select a resource, only the data that is linked to this resource is displayed in the lists for the other types of data. Irrelevant entries are filtered out.
- **New widgets**:
   - The Web Page widget displays a web page using its URL.
   - "Clock / Timer" widget.
- **Metrics Graph widget**:

>    - Added curve type format setting.

Reviewer: ```suggestion
   - Curves can now have a linear, smooth or step style.
```

   - Added multiple options to customize graphs.
   - Colors are now automatically adjusted to distinguish curves for the same metric name.
   - Enabled visualization using bar charts and stacked bar charts.
- **Resource Table widget**:
   - Added ticketing (by third-party provider) feature.
   - Added the same action buttons as in Resource Status (Acknowledge, Downtime, Forced check).
   - Added filtering options by severity and by status type.
   - The "display view" options are now available inside the widget itself.
   - Added meta-services.
- **Status Grid widget**
   - Pictograms now show the type of the resource and its state (acknowledged, in downtime).
   - A colored banner now shows the for resource's state.

**Resource Status**
- In the filter window, you can now filter by resource type.
- Users can now filter by host name in "view by host" mode.

**UX**

> Removed autocomplete for credentials fields in administration pages.

Reviewer: ```suggestion
- Removed autocomplete for credentials fields in administration pages.
```
